### PR TITLE
[RHICOMPL-922] Document use of platform basic auth in dev env

### PR DESCRIPTION
**Pull request by vkrizan**
Addition to PR #575

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -154,25 +161,11 @@ values as needed:
...
-+      f.basic_auth 'username', 'password'
-     end
-     faraday
-   end
-```
+You may also need to set up basic auth in
+[settings](config/settings/development.yml), option `platform_basic_auth`.
+This authentication will be used when talking to platform services such
+as inventory, rbac, and remediations. Anything not deployed locally will
+require basic auth instead of using an identity header (i.e. rbac, remediations):
```
Comment (akofink) [inline README.md:169]: Why is this duplicated? This applies for any development setup, so we should place it in the [Development notes](https://github.com/RedHatInsights/compliance-backend/#development-notes) section

**Review by akofink — COMMENTED** [review]
Thanks for circling back on this one! Just one comment.

File: README.md (hunk 1 above)
Comment (vkrizan) [reply README.md:169]: There isn't a shared prerequisites section, so I duplicated it. I'll figure something out.

**Review by akofink — APPROVED** [review]
Thanks, @vkrizan!

File: README.md [hunk 2]
```diff
@@ -83,6 +83,8 @@ Prerequisites:
   - `redis_url` and `redis_cache_url` configured in [settings](config/settings/development.yml)
 * URL to [Insights Inventory](https://github.com/RedHatInsights/insights-host-inventory)
   - or, `host_inventory_url` configured in [settings](config/settings/development.yml)
+* Basic authethication credenials set by
```
Comment (akofink) [inline README.md:86]:
```suggestion
* Basic authentication credentials set by
```

**Review by akofink — APPROVED** [review]
Thanks, @vkrizan!

**Pull request merged**